audobject/core/object.py

```diff
@@ -169,79 +169,40 @@ def from_dict(
             d: typing.Dict[str, typing.Any],
             root: str = None,
             **kwargs,
-    ) -> 'Object':
-        r"""Create object from dictionary.
-
-        Args:
-            d: dictionary with variables
-            root: if dictionary was read from a file, set to source directory
-            kwargs: additional variables
-
-        Returns:
-            object
-
-        Raises:
-            RuntimeError: if a mandatory argument of the object
-                is missing in the dictionary
-
-        """
-        name = next(iter(d))
-        cls, version, installed_version = utils.get_class(name)
-        params = {}
-        for key, value in d[name].items():
-            params[key] = Object._decode_value(value, **kwargs)
-        return utils.get_object(
-            cls,
-            version,
-            installed_version,
-            params,
-            root,
-            **kwargs,
+    ) -> 'Object':  # pragma: no cover
+        from audobject.core.api import load_from_dict
+        message = (
+            'audobject.Object.from_dict() is deprecated and will be removed '
+            'with version 1.0.0. Use audobject.load_from_dict() instead.'
         )
+        warnings.warn(message, category=UserWarning, stacklevel=2)
```

Comment:
Why not:
```python
@audeer.deprecated(removal_version='1.0.0', alternative='audobject.load_from_dict()')
```

Reply:
Is it supposed to work for static methods? Cause when I tried it I got an error, but will have a second look.

Reply:
Ah my bad, I only had to put the `@audeer.deprecated` decorater **after** the `@staticmethod` :) Changed it accordingly.

Reply:
Good point, should we maybe at a test that it really shows the deprecation warning?

Reply:
It's a little bit a pity with these decorators. The original idea was to make our life easier (so we don't have to write a test that the message is shown). But now we never can be sure.

I guess you tested locally at your PC?

Reply:
Yes, I tested it locally and it works. You think we should also add a test?

Reply:
No, it's ok.